backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -224,18 +234,34 @@ private Boolean doesBucketExist(String bucketName) {
           && ModelDBUtils.isEnvSet(ModelDBConstants.AWS_WEB_IDENTITY_TOKEN_FILE)) {
         // this may spiral into an infinite loop due to incorrect configuration
         LOGGER.info("Fetching temporary credentials ");
-        initializeS3ClientWithTemporaryCredentials(awsRegion, e);
+        if (initializeS3ClientWithTemporaryCredentials(awsRegion, e)) {
```

Comment:
if the s3client could not be validated continue processing the initial exception and return a 503 to the caller.
Python client retries on 503 so we do still get a few attempts to get a valid s3client , but avoid getting into a infinite loop due to incorrect configuration

Reply:
503s are very hard to debug since we have several other reasons for a 503 (e.g. envoy losing connection). Is there some other signal we can use that won't mingle with our debug signals?

Reply:
Could we use `FAILED_PRECONDITION` = 412 ??